tests/wpunit/CommentConnectionQueriesTest.php

```diff
@@ -88,184 +89,272 @@ public function create_comments() {
+		$variables['before'] = $actual['data']['comments']['pageInfo']['startCursor'];
+
+		// Set the variables to use in the WP query.
+		$query_args['offset'] = 2;
 
+		// Run the GraphQL Query
+		$expected = $wp_query->query( $query_args );
+		$actual   = $this->graphql( compact( 'query', 'variables' ) );
+
+		codecept_debug( 'expected' );
+		codecept_debug( $expected );
+
+		$this->markTestIncomplete( 'works fine until here' );
+
+		$this->assertValidPagination( $expected, $actual );
+		$this->assertEquals( true, $actual['data']['comments']['pageInfo']['hasPreviousPage'] );
+		$this->assertEquals( true, $actual['data']['comments']['pageInfo']['hasNextPage'] );
+
+		/**
+		 * Test the last two results.
+		 */
+
+		// Set the variables to use in the GraphQL query.
+		$variables['before'] = $actual['data']['comments']['pageInfo']['startCursor'];
+
+		// Set the variables to use in the WP query.
+		$query_args['offset'] = 4;
+
+		// Run the GraphQL Query
+		$expected = $wp_query->query( $query_args );
+		$expected = array_reverse( $expected );
+
+		$actual = $this->graphql( compact( 'query', 'variables' ) );
+
+		$this->assertValidPagination( $expected, $actual );
+		$this->assertEquals( false, $actual['data']['comments']['pageInfo']['hasPreviousPage'] );
+		$this->assertEquals( true, $actual['data']['comments']['pageInfo']['hasNextPage'] );
+	}
+
+	public function testForwardPagination() {
+		$this->forwardPagination();
 	}
 
 	public function testBackwardPagination() {
+		$this->backwardPagination();
+	}
+
+
+	public function testQueryWithFirstAndLast() {
+		wp_set_current_user( $this->admin );
+
+		$query = $this->getQuery();
+
+		$variables = [
+			'first' => 5,
+		];
 
 		/**
-		 * Create the cursor for the comment with the newest comment_date
+		 * Test `first`.
 		 */
-		$last_comment_cursor = \GraphQLRelay\Connection\ArrayConnection::offsetToCursor( $this->created_comment_ids[1] );
+		$actual = $this->graphql( compact( 'query', 'variables' ) );
 
+		$after_cursor  = $actual['data']['comments']['edges'][1]['cursor'];
+		$before_cursor = $actual['data']['comments']['edges'][3]['cursor'];
+
+		// Get 5 items, but between the bounds of a before and after cursor.
 		$variables = [
-			'last'   => 1,
-			'before' => $last_comment_cursor,
+			'first'  => 5,
+			'after'  => $after_cursor,
+			'before' => $before_cursor,
 		];
 
-		$results = $this->commentsQuery( $variables );
+		$expected = $actual['data']['comments']['nodes'][2];
+		$actual   = $this->graphql( compact( 'query', 'variables' ) );
 
-		$comments_query         = new WP_Comment_Query();
-		$comments               = $comments_query->query(
-			[
-				'comment_status' => 'approved',
-				'number'         => 1,
-				'offset'         => 1,
-				'order'          => 'ASC',
-				'orderby'        => 'comment_date',
-				'comment_parent' => 0,
-			]
-		);
-		$second_to_last_comment = $comments[0];
+		codecept_debug( $expected );
+		$this->markTestIncomplete( 'works until here' );
```

Comment:
@justlevine For this to work, we'll need to add functionality similar to what's been done for Post and Term connections: 

See: https://github.com/wp-graphql/wp-graphql/blob/develop/src/Data/Config.php#L187-L216